src/Bridge/Doctrine/Orm/Extension/PaginationExtension.php

```diff
@@ -121,6 +122,19 @@ public function applyToCollection(QueryBuilder $queryBuilder, QueryNameGenerator
                 $firstResult = (int) $after;
                 $firstResult = false === $after ? $firstResult : ++$firstResult;
             }
+            if (isset($collectionArgs[$resourceClass]['last'])) {
+                $firstResult = \count($queryBuilder->getQuery()->getArrayResult()) - $collectionArgs[$resourceClass]['last'];
+                $firstResult = 0 > $firstResult ? 0 : $firstResult;
+            }
+            if (isset($collectionArgs[$resourceClass]['before'])) {
+                $before = \base64_decode($collectionArgs[$resourceClass]['before'], true);
+                $firstResult = (int) $before - $itemsPerPage;
+                $firstResult = (false === $before) ? 0 : (int) $firstResult;
+                if (false !== $before && 0 > $firstResult) {
```

Comment:
I think you can remove the `false !== $before` part. Since `$firstResult` will be negative only if `$before` is not `false`.

Reply:
Is this part tested?

Reply:
This is correct I'll change that.
It is tested in features/graphql/collection.feature starting on line 424